### PR TITLE
Update module v1.3

### PR DESCRIPTION
**Pull request by tin-jy**
Add separate fields for different zoom links

Make module fields optional

Make module codes and person names case insensitive

### CODE REVIEW

File: src/main/java/seedu/address/model/module/LectureDetails.java [hunk 1]
```diff
@@ -34,7 +34,7 @@ public LectureDetails(String lectureDetails) {
      * Returns true if a given string is a valid lecture detail.
      */
     public static boolean areValidLectureDetails(String test) {
-        return test.matches(VALIDATION_REGEX);
+        return true;
```
Comment (therealdaofu) [inline src/main/java/seedu/address/model/module/LectureDetails.java:37]: Is this a temporary fix?

File: src/main/java/seedu/address/model/module/TutorialDetails.java [hunk 2]
```diff
@@ -34,7 +34,7 @@ public TutorialDetails(String tutorialDetails) {
      * Returns true if a given string is a valid tutorial detail.
      */
     public static boolean areValidTutorialDetails(String test) {
-        return test.matches(VALIDATION_REGEX);
+        return true;
```
Comment (therealdaofu) [inline src/main/java/seedu/address/model/module/TutorialDetails.java:37]: Same as above

File: src/test/java/seedu/address/logic/parser/AddModuleCommandParserTest.java [hunk 3]
```diff
@@ -4,93 +4,78 @@
...
+    //public void parse_allFieldsPresent_success() {
+    //    Module expectedModule = new ModuleBuilder(CS2103T)
+    //        .withAssignmentDetails(VALID_ASSIGNMENT_DETAILS_NORMAL).build();
+
+    //    // whitespace only preamble
+    //    assertParseSuccess(parser,
+    //        PREAMBLE_WHITESPACE + MODULE_CODE_CS2103T + LECTURE_DETAILS_CS2103T + TUTORIAL_DETAILS_CS2103T
+    //            + LECTURE_ZOOM_LINK_CS2103T + TUTORIAL_ZOOM_LINK_CS2103T + ASSIGNMENT_DETAILS_NORMAL,
+    //            new AddModuleCommand(expectedModule));
+
```
Comment (therealdaofu) [inline src/test/java/seedu/address/logic/parser/AddModuleCommandParserTest.java:36]: rmb to come back and do the test properly

**Review by therealdaofu — APPROVED** [review]
lgtm

File: src/main/java/seedu/address/model/module/LectureDetails.java (hunk 1 above)
Comment (tin-jy) [reply src/main/java/seedu/address/model/module/LectureDetails.java:37]: No it's permanent because there are no more restrictions I think

Comment (Effixion) [reply src/main/java/seedu/address/model/module/LectureDetails.java:37]: I feel like this part we need to discuss more on telegram

**Review by avock — COMMENTED** [review]
lgtm!

File: src/main/java/seedu/address/ui/ModuleCard.java [hunk 4]
```diff
@@ -56,8 +56,8 @@ public ModuleCard(Module module, int displayedIndex) {
         moduleCode.setText(module.getModuleCode().moduleCode);
         lectureDetails.setText(module.getLectureDetails().value);
         tutorialDetails.setText(module.getTutorialDetails().value);
-        lectureZoomLink.setText(module.getZoomLink().zoomLink);
-        tutorialZoomLink.setText(module.getZoomLink().zoomLink);
+        lectureZoomLink.setText(module.getLectureZoomLink().zoomLink.orElse(""));
```
Comment (avock) [inline src/main/java/seedu/address/ui/ModuleCard.java:59]: should we allow the UI class to decide whether we use an empty string? I think it should be handled by the getLectureZoomLink method

File: src/main/java/seedu/address/model/module/LectureDetails.java (hunk 1 above)
Comment (avock) [reply src/main/java/seedu/address/model/module/LectureDetails.java:37]: the initial code should remain, it detects whether the string starts with a whitespace or not

**Pull request merged**